src/System.Windows.Forms/src/System/Windows/Forms/DropTarget.cs

```diff
@@ -53,11 +66,38 @@ public DropTarget(IDropTarget owner)
                 }
             }
 
-            DragEventArgs drgevent = new DragEventArgs(data, (int)grfKeyState, pt.X, pt.Y, (DragDropEffects)pdwEffect, _lastEffect);
+            DragEventArgs drgevent = _lastDragEventArgs is null
+                ? new DragEventArgs(data, (int)grfKeyState, pt.X, pt.Y, (DragDropEffects)pdwEffect, _lastEffect)
+                : new DragEventArgs(
+                    data,
+                    (int)grfKeyState,
+                    pt.X,
+                    pt.Y,
+                    (DragDropEffects)pdwEffect,
+                    _lastEffect,
+                    _lastDragEventArgs.DropImageType,
+                    _lastDragEventArgs.Message ?? string.Empty,
+                    _lastDragEventArgs.MessageReplacementToken ?? string.Empty);
+
             _lastDataObject = data;
             return drgevent;
         }
 
+        private void ClearDropDescription()
+        {
+            _lastDragEventArgs = null;
+            DragDropHelper.ClearDropDescription(_lastDataObject);
+        }
+
+        private void UpdateDropDescription(DragEventArgs dragEventArgs)
```

Comment:
It is customary to use `e` for event args.
```suggestion
        private void UpdateDropDescription(DragEventArgs e)
```